index.bs

```diff
@@ -371,28 +404,175 @@ model]]:
+
+To <dfn>process a network cookie access for bounce tracking mitigations</dfn>
+given a [=request=] |request|, perform the following steps:
+
+1. Let |origin| be |request|'s [=request/origin=].
+1. If |origin| is an [=opaque origin=], then abort these steps.
+1. If |request|'s [=request/destination=] is "`document`", then:
+    1. [=Assert=]: |request|'s [=request/client=] is not null, and
+        |request|'s [=request/client=]'s [=environment/target browsing context=]
+        is not null.
+    1. Let |topLevelTraversable| be |request|'s [=request/client=]'s
+        [=environment/target browsing context=]'s
+        [=browsing context/top-level traversable=].
+    1. [=Assert=]: |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=] is not null.
+    1. Let |site| be the result of running [=obtain a site=] given |origin|.
+    1. [=set/Append=] |site|'s [=host=] to |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=]'s
+        [=bounce tracking record/storage access set=].
+
+Issue: TODO: Handle subresource requests and iframes for client-side redirects.
+
+</div>
+
+Note: We currently don't treat cookie reads as stateful, but this would be a
+reasonable future change. We could run [=process a network cookie access for bounce tracking mitigations=]
+in the <a spec="fetch">HTTP-network-or-cache fetch</a> algorithm after step 8.21.1.2,
+"... [=append=] (`Cookie`, cookies) to httpRequest’s [=header list=]. ..."
+
+<h5 id="bounce-tracking-mitigations-response-received-monkey-patch">Response Received Monkey
+Patch</h5>
+
+When the response is received at the end of a navigation, fill the [=bounce tracking record/bounce set=].
+
+Insert the following steps in the <a spec="html">create navigation params by fetching</a> algorithm, after Step 19.7,
+"Wait until either response is non-null...", which is the point that [=response/url list=] becomes available.
+
+1. Run [=process response received for bounce tracking=] given
+    <var ignore>navigable</var> and <var ignore>response</var>'s <var ignore>URL list</var>.
+
+<div algorithm>
+
+To <dfn>process response received for bounce tracking</dfn> given a [=navigable=]
+|navigable| and a [=list=] of [=URLs=] |URLs|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. [=Assert=]: |navigable|'s [=top-level traversable/bounce tracking record=] is not null.
+1. Let |global| be |navigable|'s [=navigable/active document=]'s [=relevant global object=].
+1. [=Run steps after a timeout=] given:
+    <dl>
+        <dt><var ignore>global</var></dt>
+        <dd>|global|</dd>
+        <dt><var ignore>milliseconds</var></dt>
+        <dd>[=client bounce detection timer period=]</dd>
+        <dt><var ignore>completionSteps</var></dt>
+        <dd>[=queue a global task=] on the [=networking task source=] with |global| to [=record stateful bounces for bounce tracking=] given |global|</dd>
+    </dl>
```

Comment:
This will accumulate a queue of [=record stateful bounces=] tasks as the browser follows client-side redirects, rather than only running once for the last redirect. The logic that checks if the global's navigable's document is still the global's document might abort the other ones early, but please double-check that and make sure it's doing what you intend.